vault/vault.go

```diff
@@ -193,6 +225,79 @@ func (v *vaultSecrets) ListSecrets() ([]string, error) {
 	return nil, secrets.ErrNotSupported
 }
 
+func (v *vaultSecrets) read(path string) (*api.Secret, error) {
+	secretValue, err := v.lockedRead(path)
+	if v.isTokeExpired(err) {
+		if err = v.renewToken(); err != nil {
+			return nil, err
+		}
+		return v.lockedRead(path)
+	}
+	return secretValue, err
+}
+
+func (v *vaultSecrets) write(path string, data map[string]interface{}) (*api.Secret, error) {
+	v.mu.RLock()
```

Comment:
no need of Rlock here as lockedWrite is anyways taking a Rlock

Reply:
Updated.